src/icu4c.mk

```diff
@@ -20,35 +19,42 @@ define $(PKG)_UPDATE
 endef
 
 define $(PKG)_BUILD_COMMON
-    cd '$(1)/source' && autoreconf -fi
-    mkdir '$(1).native' && cd '$(1).native' && '$(1)/source/configure' \
+    cd '$(SOURCE_DIR)/source' && autoreconf -fi
+    mkdir '$(BUILD_DIR)/$(PKG).native'
+    cd '$(BUILD_DIR)/$(PKG).native' && '$(SOURCE_DIR)/source/configure' \
         CC=$(BUILD_CC) CXX=$(BUILD_CXX)
-    $(MAKE) -C '$(1).native' -j '$(JOBS)'
+        $(MAKE) -C '$(BUILD_DIR)/$(PKG).native' -j $(JOBS) VERBOSE=1
 
-    mkdir '$(1).cross' && cd '$(1).cross' && '$(1)/source/configure' \
+    mkdir '$(BUILD_DIR)/$(PKG).cross'
+    cd '$(BUILD_DIR)/$(PKG).cross' && '$(SOURCE_DIR)/source/configure' \
         $(MXE_CONFIGURE_OPTS) \
-        --with-cross-build='$(1).native' \
-        CFLAGS=-DU_USING_ICU_NAMESPACE=0 \
-        CXXFLAGS='--std=gnu++0x' \
-        SHELL=bash
-
-    $(MAKE) -C '$(1).cross' -j '$(JOBS)' install
+        --with-cross-build='$(BUILD_DIR)/$(PKG).native' \
+        CFLAGS='-DU_USING_ICU_NAMESPACE=0' \
+        CXXFLAGS='-std=c++11' \
+        LIBS='-lmsvcr100' \
+        SHELL=$(SHELL)
+    
+    rm -fv '$(PREFIX)/$(TARGET)/lib/libicu*.dll.a'
```

Comment:
Check your shell quoting, this does nothing.

Reply:
Except it does, I'll explain below.

Reply:
Without adding shell like this, build fails with an obscure error on some systems because of different `echo` implementations (Support for `-n`), on some systems we end uo generating broken makefiles if `sh` is used instead of bash.

Reply:
I understand the `SHELL=$(SHELL)`, we've seen that often in the past.

I meant this line:
```
rm -fv '$(PREFIX)/$(TARGET)/lib/libicu*.dll.a'
```

will try to delete a file literally named `libicu*.dll.a`.

Reply:
Ah. You're right.
Still, I think this one is not quite ready for primetime yet. The thing that is puzzling me right now... I've swiped some of the msys2 patches to install dlls in bin; however I cannot understand why it installs dlls in bin (including a stub data) and import libs in lib, where it also places the full data dll and a dll.a for the stub data file...

I just don't understand that.
I'm taking a look at the makefiles and it seems the full data lacks logic for creating/installing the import library that the stubdata has. Do you know what could be the reasoning behind this?

Reply:
the *.dll files are used for deployment and need to be in the runtime directory to be found without a modifying %PATH%. The *.dll.a are used for linking and will usually be looked for in `lib`. 

I'm guessing they are trying to default to linking against the stub data so you can have more flexibility at runtime?

Reply:
> the *.dll files are used for deployment and need to be in the runtime directory to be found without a modifying %PATH%. The *.dll.a are used for linking and will usually be looked for in lib.

That I understand.

> I'm guessing they are trying to default to linking against the stub data so you can have more flexibility at runtime?

I'm guessing that _could_ make sense? Excpt right now it's the other way around (stub in bin, full in lib; although that could partially be my fault)

Anyway, what do you think we should do? Keep the way you say (stub for linking, full data in bin)? Or should we have full data both for runtime and linking? I'd say the latter but I don't know.

Reply:
The latter sounds more reasonable as a default, using the stub data could lead to unexpected surprises when deploying.

I'll see if there's a simple test program to try.

Reply:
Would we necessarily need an import lib or is it feasible to just drop the dll (or a symlink to it) in `lib`?

As for test programs... havent' actually tested it yet but I got one from a blog...

Reply:
https://gist.github.com/Lord-Kamina/9b8dbe079c58420ef9850f3922231740

Reply:
Thanks, I've added that https://github.com/mxe/mxe/pull/2421/commits/2ec080e7e05e86c91976c22367458899d6685317 in #2421

It builds a zip that packages the correct dlls and executes on Win7 using msvcrt.